### ISSUE TITLE
With a trial period enabled, after subscribing via Stripe payment method, the payment method is not saved in my-account

### ISSUE THREAD

**Issue by james-allan** [woocommerce/woocommerce-gateway-stripe#1611]
This is a copy of an issue originally posted to WC Subscriptions 4120-gh-woocommerce/woocommerce-subscriptions by @grig-23 

-----

### Describe the bug
With a trial period enabled, after subscribing via [Stripe](https://woocommerce.com/products/stripe/) payment method, the payment method is not saved in my-account.

In other words, if someone subscribes to a product that has a free trial period or uses an additional extension like [this one](https://woocommerce.com/products/custom-start-date-for-woocommerce-subscriptions/) - the customer ID and source ID will be saved in the billing details (wp-admin), however, there will be no saved payment method in my-account (front end).

### To reproduce

1. Subscription product with a free trial.
2. Incognito mode as a new customer - subscribe using Stripe (sandbox mode works). Use a test card 4242424242424242.
3. Subscription will renew correctly in the future (customer/source IDs are present), however, there isn't a payment method added in my-account >> payment methods

**Screenshots**

![Screenshot](https://d.pr/i/WbrqE2+)
Screenshot direct link: https://d.pr/i/WbrqE2

An example with Custom Start Date:

https://d.pr/f/kKGHsW

### Expected behavior
There should be a saved payment method in my-account.

User says it started happening after one of the recent updates. Same thing happens with Subscriptions 3.1.2 and Stripe 5.2.0 though.

### Additional details
Reported in 4086666-zen
With a trial period enabled, after subscribing via Stripe payment method, the payment method is not saved in my-account #4120

**Comment by christianblueskylabs** [comment]
I have a similar issue but it's not from a subscription with a trial period enabled.

I had a chat with WooCommerce Support where I was asking if payment methods (stripe tokens in this case) were being saved in the site when changing payment methods on an on-going subscription. I was told that this is a yes, but this was not happening to me with the same version as yours. I was told that the issue I might be having is related to this, so I was hoping that our issues might be related as we have the same side effect we're experiencing. Are you able to confirm this on your side as well?

**Comment by RFMENNECIER** [comment]
Hello
I was adviced to wait for a fix or to downgrade the plugin to version 5.1 to fix the problem.
I used this extension to downgrade the stripe plugin : https://fr.wordpress.org/plugins/wp-rollback/

Hope this helps ??

De : ***@***.***>
Envoyé le :jeudi 24 juin 2021 18:00
À : ***@***.***>
Cc : ***@***.******@***.***>; ***@***.***>
Objet :Re: [woocommerce/woocommerce-gateway-stripe] With a trial period enabled, after subscribing via Stripe payment method, the payment method is not saved in my-account (#1611)


I have a similar issue but it's not from a subscription with a trial period enabled.

I had a chat with WooCommerce Support where I was asking if payment methods (stripe tokens in this case) were being saved in the site when changing payment methods on an on-going subscription. I was told that this is a yes, but this was not happening to me with the same version as yours. I was told that the issue I might be having is related to this, so I was hoping that our issues might be related as we have the same side effect we're experiencing. Are you able to confirm this on your side as well?

**Comment by RFMENNECIER** [comment]
> I have a similar issue but it's not from a subscription with a trial period enabled.
> 
> I had a chat with WooCommerce Support where I was asking if payment methods (stripe tokens in this case) were being saved in the site when changing payment methods on an on-going subscription. I was told that this is a yes, but this was not happening to me with the same version as yours. I was told that the issue I might be having is related to this, so I was hoping that our issues might be related as we have the same side effect we're experiencing. Are you able to confirm this on your side as well?

Hello
I was advised to wait for a fix or to downgrade the plugin to version 5.1 to fix the problem.
I used this extension to downgrade the stripe plugin: https://fr.wordpress.org/plugins/wp-rollback/

Hope this helps ??

**Comment by christianblueskylabs** [comment]
@RFMENNECIER that definitely helped for now. Thank you!

**Comment by AashikP** [comment]
I can also replicate this behavior with the following steps:

1. Create two Subscription products: (The only difference is with signup Fee)

    ![4qgcOq.png](https://user-images.githubusercontent.com/17475174/193981224-1abc0d19-7d0e-4f39-8feb-3022d746fc4e.png)
    
    ![HnmgrJ.png](https://user-images.githubusercontent.com/17475174/193981298-fbbddd8f-67f2-422f-9504-c511a3da814f.png)
    

2. Disable $0 checkout and manual renewals (may be replicable with those enabled as well) under Subscription settings
3. Checkout the product with _no Signup Fee_ as a new user to confirm that the card is not saved under Payment methods
4. The renewal still goes through successfully 


Reported in 5481773-zen

**Comment by Babylon1999** [comment]
zd-6563675 mentions they were able to fix this by enabling the "New Checkout Experience"

**Comment by scottbuscemi** [comment]
Seeing the same. @james-allan How can I help get this prioritized for a fix? Thank you 🙏🏼

**Comment by james-allan** [comment]
> zd-6563675 mentions they were able to fix this by enabling the "New Checkout Experience"

Yep that's correct, this issue is an old issue and is limited to stores using the Stripe legacy checkout experience. Disabling it will resolve the issue. 

FWIW, Stripe are moving to deprecate and sunset the APIs that are used by the legacy checkout experience and so merchants should be actively migrating away from that setting.

---

📓 **_Once support for the legacy checkout is dropped, this issue can be closed._**

**Comment by wjrosa** [comment]
Closing as obsolete. This was reported against the legacy Sources-based checkout, which has been removed; UPE is now always enabled and free-trial purchases save the payment method through SetupIntents. If the saved method still doesn't appear under My Account on a current version, please open a fresh issue.

> *Drafted with AI; reviewed by me.*
